docs/source/benchmarking/performance.rst

```diff
@@ -181,3 +181,21 @@ read `this <https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zer
 
         def optimizer_zero_grad(self, epoch, batch_idx, optimizer, optimizer_idx):
             optimizer.zero_grad(set_to_none=True)
+
+----------
+
+Skip validation for early epochs
+------------------------------
+
+If you don't want to perform validation for early epochs, i.e., for ``epoch < num_no_val_epochs``, you can disable validation using the ``on_train_epoch_start`` hook. This can be especially useful when validation is performed after each training batch.
+
+Note: doesn’t work nicely with ``EarlyStopping``.
+
+.. code-block:: python
+
+    def on_train_epoch_start(self):
+        if self.trainer.current_epoch >= self.num_burnin_epochs:
+            self.trainer.limit_val_batches = 1.0
+        else:
+            self.trainer.limit_val_batches = 0
+
```

Comment:
this can be quite risky updating the trainer properties like this, especially if the trainer derives state from these arguments before the module changes them here. it can quickly result in things falling out of sync. 

@awaelchli @carmocca @justusschock : this is supported on the training batch loop, but not the evaluation loop: https://github.com/PyTorchLightning/pytorch-lightning/blob/d4d5418cc4086a7cb8ecc18a42e0eb109b3b7859/pytorch_lightning/loops/training_batch_loop.py#L81-L91

any reason why? if the lightning module had a way to signal to the trainer to short-circuit the val/test epoch, resetting the trainer properties wouldn't be needed.

Reply:
for this usecase it feels more natural to introduce an enable/disable mechanism on the loops, for example:

```
# could be method or property setter
self.trainer.fit_loop.validation_loop.disable = self.trainer.current_epoch < self.num_burnin_epochs
```

because if the intention is to skip the val loop entirely, then returning -1 from on_val_batch_start is not fully satisfactory as the on_validation_epoch_start / end etc. hooks would still run.

Reply:
Thanks for your inputs. Do you think it's still worth mentioning this trick in the doc ?

Reply:
There's no need for a disable mechanism if loops are customizable, right?

Reply:
The question is whether we want to recommend the user to implement/subclass a whole new loop class just to add a disable condition to it, vs. exposing a method.